lib/puppet/util/log.rb

```diff
@@ -175,7 +175,7 @@ def Log.coerce_string(str)
     # We only select the last 10 callers in the stack to avoid being spammy
     message = _("Received a Log attribute with invalid encoding:%{log_message}") %
         { log_message: Puppet::Util::CharacterEncoding.convert_to_utf_8(str.dump)}
-    message += '\n' + _("Backtrace:\n%{backtrace}") % { backtrace: caller[0..10].join("\n") }
+    message += '\n' + _("Backtrace:\n%{backtrace}") % { backtrace: caller(1, 10).join("\n") }
```

Comment:
👍 previously there was an off-by-one error (we returned 11 frames instead of 10 as specified in the comments).